tests/src/JIT/Stress/ABI/ABIs.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Text;
+
+namespace ABIStress
+{
+    internal interface IAbi
+    {
+        Type[] TailCalleeCandidateArgTypes { get; }
+        CallingConvention[] PInvokeConventions { get; }
+        int ApproximateArgStackAreaSize(List<TypeEx> parameters);
+    }
+
+    internal class Win86Abi : IAbi
+    {
+        public Type[] TailCalleeCandidateArgTypes { get; } =
+            new[]
+            {
+                typeof(byte), typeof(short), typeof(int), typeof(long),
+                typeof(float), typeof(double),
+                typeof(Vector<int>), typeof(Vector128<int>), typeof(Vector256<int>),
+                typeof(S1P), typeof(S2P), typeof(S2U), typeof(S3U),
+                typeof(S4P), typeof(S4U), typeof(S5U), typeof(S6U),
+                typeof(S7U), typeof(S8P), typeof(S8U), typeof(S9U),
+                typeof(S10U), typeof(S11U), typeof(S12U), typeof(S13U),
+                typeof(S14U), typeof(S15U), typeof(S16U), typeof(S17U),
+                typeof(S31U), typeof(S32U),
+            };
+
+        public CallingConvention[] PInvokeConventions { get; } = { CallingConvention.Cdecl, CallingConvention.StdCall, CallingConvention.ThisCall, };
+
+        public int ApproximateArgStackAreaSize(List<TypeEx> parameters)
+        {
+            int size = 0;
+            foreach (TypeEx pm in parameters)
+                size += (pm.Size + 3) & ~3;
```

Comment:
Can you add a roundup method and explain what this is accomplishing. Was hard to follow just reading through the code.